modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridNearAtomicUpdateFuture.java

```diff
@@ -668,7 +673,12 @@ private void sendUpdateRequests(Map<UUID, PrimaryRequestState> mappings) {
                     if (req.initMappingLocally() && reqState.mappedNodes.isEmpty())
                         reqState.resetLocalMapping();
 
-                    cctx.io().send(req.nodeId(), req, cctx.ioPolicy());
+                    GridCacheMessage msg = req;
```

Comment:
Let's create method to wrap msg into AtomicApplicationAttributesAwareRequest if required.